File: src/main/java/com/wesker926/leetcode/algorithms/p0383/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0383.s1;

/**
 * @author wesker.gh
 * @date 2021/12/4
 * @description 字母表
 */
public class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        if (magazine.length() < ransomNote.length()) {
            return false;
        }

        int[] map = new int[26];
        for (char c : magazine.toCharArray()) {
            map[c - 'a']++;
        }

        for (char c : ransomNote.toCharArray()) {
            if ((map[c - 'a']--) <= 0) {
                return false;
            }
        }
        return true;
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0383/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/12/4
// @description: 字母表
package s1

func canConstruct(ransomNote string, magazine string) bool {
	if len(magazine) < len(ransomNote) {
		return false
	}

	table := make([]int, 26)
	for _, c := range magazine {
		table[int(c-'a')]++
	}

	for _, c := range ransomNote {
		if table[int(c-'a')]--; table[int(c-'a')] < 0 {
			return false
		}
	}
	return true
}
